src/Components/AdvancedSearch/Internal/ColumnPicker/ColumnPicker.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import {
+    IColumnPickerProps,
+    IColumnPickerStyleProps,
+    IColumnPickerStyles
+} from './ColumnPicker.types';
+import { getStyles } from './ColumnPicker.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    Icon,
+    Dropdown,
+    IDropdownOption
+} from '@fluentui/react';
+import { useTranslation } from 'react-i18next';
+
+const getClassNames = classNamesFunction<
+    IColumnPickerStyleProps,
+    IColumnPickerStyles
+>();
+
+const ColumnPicker: React.FC<IColumnPickerProps> = ({
+    searchedProperties,
+    allAvailableProperties,
+    addColumn,
+    deleteColumn,
+    styles
+}) => {
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+    const { t } = useTranslation();
+    const [selectedKeys, setSelectedKeys] = React.useState<string[]>([]);
+    const [selectedPropsCount, setSelectedPropsCount] = React.useState(
```

Comment:
let's remove this state and simply reference `selectedKeys.length` so we don't need extra renders.